tooling/hibernate-maven-plugin/hibernate-maven-plugin.gradle

```diff
@@ -31,6 +31,8 @@ dependencies {
 
     compileOnly "org.apache.maven.plugin-tools:maven-plugin-tools-annotations:3.15.1"
 
+    testRuntimeOnly 'org.apache.maven:maven-core:3.9.11'
```

Comment:
```
    implementation "org.apache.maven:maven-plugin-api:3.9.11"
    implementation "org.apache.maven:maven-core:3.9.11"
    implementation "org.apache.maven.shared:file-management:3.1.0"
```
hmm I didn't want to add that to your list initially 🙂 but since you ended up needing the `maven-core` for tests... 
I think we should just go ahead and replace the `maven-project` (line 29). it's part of the maven core and with that ^ we can also remove the unchecked suppression from the Mojo class (since if we work with the `MavenProject` from core, those artifacts set is nicely typed 🙂).

the config for the publishing task below (`publishedArtifacts`) should take care and apply the `provided` scope to the published plugin pom.